modules/installation-osp-creating-network-resources.adoc

```diff
@@ -0,0 +1,671 @@
+      protocol: tcp
+      remote_group: "{{ os_sg_worker }}"
+      port_range_min: 30000
+      port_range_max: 32767
+
+  - name: 'Create worker-sg rule "worker ingress services (TCP) from master"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: tcp
+      remote_group: "{{ os_sg_master }}"
+      port_range_min: 30000
+      port_range_max: 32767
+
+  - name: 'Create worker-sg rule "worker ingress services (UDP)"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: udp
+      remote_group: "{{ os_sg_worker }}"
+      port_range_min: 30000
+      port_range_max: 32767
+
+  - name: 'Create worker-sg rule "worker ingress services (UDP) from master"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: udp
+      remote_group: "{{ os_sg_master }}"
+      port_range_min: 30000
+      port_range_max: 32767
+
+  - name: 'Create worker-sg rule "VRRP"'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: '112'
+      remote_ip_prefix: "{{ os_subnet_range }}"
+----
+. Insert the following content into a local file called `02_network.yaml`
++
+[source,yaml]
+----
+# Required Python packages:
+#
+# ansible
+# openstacksdk
+# netaddr
+# openstackclient
+
+- import_playbook: common.yaml
+
+- hosts: all
+  gather_facts: no
+
+  tasks:
+  - name: 'Create the cluster network'
+    os_network:
+      name: "{{ os_network }}"
+
+  - name: 'Set the cluster network tag'
+    os_tag:
+      resource: "network"
+      name: "{{ os_network }}"
+      tags: "{{ [cluster_id_tag] }}"
+
+  - name: 'Create a subnet'
+    os_subnet:
+      name: "{{ os_subnet }}"
+      network_name: "{{ os_network }}"
+      cidr: "{{ os_subnet_range }}"
+      allocation_pool_start: "{{ os_subnet_range | next_nth_usable(10) }}"
+      allocation_pool_end: "{{ os_subnet_range | ipaddr('last_usable') }}"
+
+  - name: 'Set the cluster subnet tag'
+    os_tag:
+      resource: "subnet"
+      name: "{{ os_subnet }}"
+      tags: "{{ [cluster_id_tag] }}"
+
+  - name: 'Create the service network'
+    os_network:
+      name: "{{ os_svc_network }}"
+    when: os_networking_type == "Kuryr"
+
+  - name: 'Set the service network tag'
+    os_tag:
+      resource: "network"
+      name: "{{ os_svc_network }}"
+      tags: "{{ [cluster_id_tag] }}"
+    when: os_networking_type == "Kuryr"
+
+  - name: 'Create the service subnet'
+    os_create_svc_subnet:
+      subnet_name: "{{ os_svc_subnet }}"
+      network_name: "{{ os_svc_network }}"
+      cidr: "{{ svc_subnet_range }}"
+      openstack_svc_network: "{{ os_svc_network_range }}"
+    when: os_networking_type == "Kuryr"
+
+  - name: 'Set the service subnet tag'
+    os_tag:
+      resource: "subnet"
+      name: "{{ os_svc_subnet }}"
+      tags: "{{ [cluster_id_tag] }}"
+    when: os_networking_type == "Kuryr"
+
+  - name: 'Create pods subnet pool'
+    os_create_subnet_pool:
```

Comment:
We have a problem here.

These modules:
* os_create_subnet_pool
* os_create_svc_subnet
* os_delete_lb_resources
* os_delete_network_resources
* os_delete_ports_on_router
* os_tag
* os_trunk

are not available in Ansible out-of-the-box. We coded them, and they are available in https://github.com/openshift/installer/tree/master/upi/openstack/library

Ansible will load all modules available in `./library`, but this means that you have to run Ansible from `${installer_repository_folder}/upi/openstack`. You can't just copy the content of the modules in a new file and expect them to run from anywhere in the system.

Can we ask the user to clone the repo and run from there?

Reply:
@pierreprinetti We're disallowed from linking to anything on GitHub in the downstream docs for now. It is not a fun restriction. We're doing our best to get past it--possibly by providing an RPM to customers--but Support is concerned about testing and reliability.

Reply:
One option is to insert all of these modules into the ShiftStack UPI docs and direct the user to do the whole copy-save dance that we ask them to do for the Ansible playbooks. 

@aheslin (and Stephanie--can't find you with @-search but I'll send a Slack DM)--do you have any opinions about this? The process is already demanding a lottt of file transfer via clipboard.

Reply:
In my opinion copying files via clipboard will be dangerous and error prone. I think just clone it from a repo is much safer (and of course have an rpm is even better)

Reply:
Couldn't we rely on RPM here?

Reply:
@MaysaMacedo  You mean packaging our additional modules (`os_create_subnet_pool` etc) as RPMs? Or are they available as RPMs already?

Reply:
I'm strongly in favor of doing this via RPM. If that's something dev can push Support on, I'd be happy.